code/modules/power/cell.dm

```diff
@@ -118,9 +118,7 @@
 		rigged = 1 //broken batterys are dangerous
 
 /obj/item/weapon/cell/emp_act(severity)
-	charge -= 1000 / severity
-	if (charge < 0)
-		charge = 0
+	charge = max(charge - maxcharge/severity, 0)
```

Comment:
Yes that is how the CURRENT formula works, I'm looking at ways to change it so it does about 25-50% for 1/2 at 1000, and about 10-20% at 1/2 on a hyper cell.